test/boost/secondary_index_test.cc

```diff
@@ -1984,6 +1984,129 @@ SEASTAR_TEST_CASE(test_deleting_ghost_rows) {
     });
 }
 
+SEASTAR_TEST_CASE(test_deleting_ghost_rows_with_same_base_pk) {
```

Comment:
I must say that it is weird that those tests are in `secondary_index_test.cc` while they are about testing the functionality of materialized views. However, other tests for this feature are in the same file as well, so I won't insist on moving anything in this PR.